app/models/user.rb

```diff
@@ -22,21 +22,32 @@
 #
 class User < ApplicationRecord
 
+  # TODO: Move this to Global Application Config class
+  THEMES ||= %w[system dark light].freeze
+
+  after_initialize :set_default_preferences, if: :new_record?
   before_save :sanitize_names
 
+  serialize :preferences, coder: HashSerializer
+  store_accessor :preferences, :theme, prefix: :preference
+  validates :preferences, preferences: true
```

Comment:
This line looks a bit weird but its fine. Can't think of a better name

Reply:
Agree looks strange but that's how it is done with `EachValidator` unfortunately

https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations